test/framework/fileConversion.go

```diff
@@ -1,35 +1,73 @@
 package framework
 
 import (
-	"os"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+
+	"github.com/kubevirt/containerized-data-importer/pkg/image"
 )
 
-const BaseImageFile = "../image/tinyCore.iso" // relative path to image file
+// createGz compresses a given test file and stores it to the target
+func CreateGz(src []byte) ([]byte, error) {
```

Comment:
what about `exec`ing the _tar vvfz_ and _gzip_ commands rather than creating this file formats using go code? I'm guessing one reason is to keep the test buffer in memory. But a downside is that there could potentially be mirror bugs in the go compress or archive packages such that the file compresses and decompress correctly using the same pkg but would not decompress had the file been created from the shell. (Of course we're not really expected to test a golang pkg but we are expected to decompress and dearchive files correctly.)
Also, not sure if this will work as smoothly for _xz_.

Reply:
The trouble with writing to disk is that the compression binaries won't be consistent across environments.   One solution would be to containerize the tests, which I think is reasonable.  We'd also have a consistent golang version, which adds further stability.